### ISSUE TITLE
nt.predict.gradient_descent fails with vmap

### ISSUE THREAD

**Issue by tancik** [google/neural-tangents#32]
Should we expect nt.predict.gradient_descent to fail when using jax vmap due to the scipy ode solver? Are there any suggested workarounds for speeding this function up over batches?

**Comment by romanngg** [comment]
Thanks for reporting these issues - we have some WIP changes to the predict module that I hope will make it both faster and less error-prone.

Just to be clear here, could you post a code snippet of how you're vmapping and what error you're getting?

**Comment by tancik** [comment]
Here is a minimal version of the code:
```
from jax import random, vmap
from neural_tangents import stax
import neural_tangents as nt

rand_key = random.PRNGKey(10)
data_x = np.linspace(-1,1,256)[...,None]
data_y = random.normal(rand_key, (256,1))

init_fn, apply_fn, kernel_fn = stax.serial(stax.Dense(512), stax.Relu(), stax.Dense(1))

g_dd = kernel_fn(data_x, data_x, 'ntk')
loss_fn = lambda fx, y_hat: np.mean(fx - y_hat)
train_predict_fn = nt.predict.gradient_descent(g_dd, data_y, loss_fn)

_, params = init_fn(rand_key, (256, 1))
fx_train = apply_fn(params, data_x)

train_predict_fn(5, fx_train) ## Works

vmap(train_predict_fn, in_axes=(0, None))(np.arange(1000), fx_train) ## Breaks
```

The error returned is:
`SystemError: _dop.dopri5() 4th argument (xend) can't be converted to double`

I am new to JAX so it is possible I am misunderstanding how vmap works.

**Comment by romanngg** [comment]
Thanks for the example and again, sorry for the delay - fixed in a76bbb494f19af4f8c9c1a1b0904e91b105f769e.

Here's your example working:
https://colab.research.google.com/gist/romanngg/241db30f1da848c5a2eb89f31f2218c7/predict_vmap.ipynb

Note at the bottom, that we now support natively array-valued, of any shape times `t` - so you can just pass time arrays to the function without vmap!
